### PR TITLE
Don't throw NPE if topic is not specified when creating a stream / table

### PR DESCRIPTION
**Pull request by mitch-seymour**
### Description 
Fixes https://github.com/confluentinc/ksql/issues/2138. 

A `NullPointerException` is thrown if an `AVRO` stream / table is created without the mandatory `KAFKA_TOPIC` parameter.

```
ksql> CREATE TABLE FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='AVRO');
java.lang.NullPointerException

ksql> CREATE STREAM FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='AVRO');
java.lang.NullPointerException
```

There is, however, a proper error message for `JSON` streams / tables that are created without the `KAFKA_TOPIC` parameter:

```
ksql> CREATE TABLE FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='JSON');
Corresponding kafka topic(KAFKA_TOPIC) should be set in WITH clause.

ksql> CREATE STREAM FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='JSON');
Corresponding kafka topic(KAFKA_TOPIC) should be set in WITH clause.
```

This PR ensures the above error is returned for `AVRO` streams as well, and makes a slight change to the error message (adds a missing space).

### Testing done 
Created and ran (via mvn) the new test included in this PR. Also tested manually.

### Reviewer checklist
- [x] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [x] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by mitch-seymour** [comment]
[clabot:check]

**Comment by mitch-seymour** [comment]
FWIW, here's the output of the manual test I ran after making this change, which shows the updated output of various CREATE statements without the mandatory `KAFKA_TOPIC` parameter.

```
ksql> CREATE TABLE FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='AVRO');
Corresponding kafka topic (KAFKA_TOPIC) should be set in WITH clause.

ksql> CREATE STREAM FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='AVRO');
Corresponding kafka topic (KAFKA_TOPIC) should be set in WITH clause.

ksql> CREATE TABLE FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='JSON');
Corresponding kafka topic (KAFKA_TOPIC) should be set in WITH clause.

ksql> CREATE STREAM FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='JSON');
Corresponding kafka topic (KAFKA_TOPIC) should be set in WITH clause.
```

Of course, this is also captured in the

File: ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/AbstractCreateStreamCommand.java [hunk 1]
```diff
@@ -172,9 +172,10 @@ private RegisterTopicCommand registerTopicFirst(
           "Topic format(" + DdlConfig.VALUE_FORMAT_PROPERTY + ") should be set in WITH clause.");
     }
     if (!properties.containsKey(DdlConfig.KAFKA_TOPIC_NAME_PROPERTY)) {
-      throw new KsqlException(
-          "Corresponding kafka topic(" + DdlConfig.KAFKA_TOPIC_NAME_PROPERTY
-          + ") should be set in WITH clause.");
+      throw new KsqlException(String.format(
+          "Corresponding kafka topic (%s) should be set in WITH clause.",
```
Comment (rmoff) [inline ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/AbstractCreateStreamCommand.java]: Nit: `Kafka topic` not `kafka topic`

File: ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java [hunk 2]
```diff
@@ -63,6 +63,12 @@ public static AbstractStreamCreateStatement checkAndSetAvroSchema(
       return abstractStreamCreateStatement;
     }
 
+    if (!ddlProperties.containsKey(DdlConfig.KAFKA_TOPIC_NAME_PROPERTY)) {
+      throw new KsqlException(String.format(
+          "Corresponding kafka topic (%s) should be set in WITH clause.",
```
Comment (rmoff) [inline ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java]: Nit: `Kafka topic` not `kafka topic`

File: ksql-engine/src/test/java/io/confluent/ksql/QueryEngineTest.java [hunk 3]
```diff
@@ -89,4 +89,26 @@ public void shouldThrowExpectedExceptionForDuplicateStream() {
...
+      "CREATE STREAM FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='JSON');",
+    };
+    for (String statement : statements) {
+      try {
+        final List<PreparedStatement> statementList = ksqlEngine.parseStatements(
+            statement, metaStore.clone(), true);
+        queryEngine.buildLogicalPlans(metaStore, statementList, ksqlConfig);
+        Assert.fail();
+      } catch (final KsqlException e) {
+        assertThat(e.getMessage(), equalTo("Exception while processing statement: Corresponding kafka topic (KAFKA_TOPIC) should be set in WITH clause."));
```
Comment (rmoff) [inline ksql-engine/src/test/java/io/confluent/ksql/QueryEngineTest.java]: Nit: `Kafka topic` not `kafka topic`

Comment (mitch-seymour) [reply ksql-engine/src/test/java/io/confluent/ksql/QueryEngineTest.java]: Fixed :)

**Comment by mitch-seymour** [comment]
@rmoff I fixed the casing in the error message :) See here:

```
ksql> CREATE TABLE FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='AVRO');
Corresponding Kafka topic (KAFKA_TOPIC) should be set in WITH clause.

ksql> CREATE STREAM FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='AVRO');
Corresponding Kafka topic (KAFKA_TOPIC) should be set in WITH clause.

ksql> CREATE TABLE FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='JSON');
Corresponding Kafka topic (KAFKA_TOPIC) should be set in WITH clause.

ksql> CREATE STREAM FOO (viewtime BIGINT, pageid VARCHAR) WITH (VALUE_FORMAT='JSON');
Corresponding Kafka topic (KAFKA_TOPIC) should be set in WITH clause
```

**Comment by big-andy-coates** [comment]
LGTM, thanks @mitch-seymour

**Comment by mitch-seymour** [comment]
looks like CI checks are passing :) feel free to merge at will. Thanks for the quick review, all!

**Pull request merged**